cnf-tests/testsuites/e2esuite/dpdk/dpdk.go

```diff
@@ -1134,6 +1186,122 @@ func findNUMAForSRIOV(pod *corev1.Pod) (int, error) {
 	return -1, fmt.Errorf("failed to find the numa for pci %s", pciEnvVariableName)
 }
 
+func checkDownwardApi(pod *corev1.Pod, path string, c string) (string, error) {
+	var output string
+
+	switch {
+	case c == "label":
+		buff, err := pods.ExecCommand(client.Client, *pod, []string{"cat", "/etc/podnetinfo/" + path})
+		if err != nil {
+			fmt.Println("Volume", path, "is not present")
+		}
+		output = buff.String()
+	case c == "IP":
+		buff, err := pods.ExecCommand(client.Client, *pod, []string{"cat", "/etc/podnetinfo/" + path})
+		if err != nil {
+			fmt.Println("Volume", path, "is not present")
```

Comment:
In case of an error we exit with an empty buff, and will fail in one of the  Expect().To(ContainSubstring()) statements.
Maybe it would be more clear if we return err here, and Expect Err NotToOccur outside the function?

Reply:
I had a look at the other test cases in dpdk which used the same function and removed the if err != nil and added the  Expect Err NotToOccur